storage/src/test/java/org/apache/kafka/server/log/remote/storage/LocalTieredStorage.java

```diff
@@ -0,0 +1,564 @@
+        if (files == null) {
+            // files can be null if the directory is empty.
+            return;
+        }
+
+        Arrays.stream(files)
+                .filter(File::isDirectory)
+                .forEach(dir ->
+                        openExistingTopicPartitionDirectory(dir.getName(), storageDirectory).traverse(traverser));
+    }
+
+    public void addListener(final LocalTieredStorageListener listener) {
+        this.storageListeners.add(listener);
+    }
+
+    @Override
+    public void configure(Map<String, ?> configs) {
+        if (storageDirectory != null) {
+            throw new InvalidConfigurationException(format("This instance of local remote storage" +
+                    "is already configured. The existing storage directory is %s. Ensure the method " +
+                    "configure() is only called once.", storageDirectory.getAbsolutePath()));
+        }
+
+        final String storageDir = (String) configs.get(STORAGE_DIR_PROP);
+        final String shouldDeleteOnClose = (String) configs.get(DELETE_ON_CLOSE_PROP);
+        final String transfererClass = (String) configs.get(TRANSFERER_CLASS_PROP);
+        final String isDeleteEnabled = (String) configs.get(ENABLE_DELETE_API_PROP);
+        final Integer brokerIdInt = (Integer) configs.get(BROKER_ID);
+
+        if (brokerIdInt == null) {
+            throw new InvalidConfigurationException(
+                    "Broker ID is required to configure the LocalTieredStorage manager.");
+        }
+
+        brokerId = brokerIdInt;
+        logger = new LogContext(format("[LocalTieredStorage Id=%d] ", brokerId)).logger(this.getClass());
+
+        if (shouldDeleteOnClose != null) {
+            deleteOnClose = Boolean.parseBoolean(shouldDeleteOnClose);
+        }
+
+        if (isDeleteEnabled != null) {
+            deleteEnabled = Boolean.parseBoolean(isDeleteEnabled);
+        }
+
+        if (transfererClass != null) {
+            try {
+                transferer = (Transferer) getClass().getClassLoader().loadClass(transfererClass).newInstance();
+
+            } catch (InstantiationException | IllegalAccessException | ClassNotFoundException | ClassCastException e) {
+                throw new RuntimeException(format("Cannot create transferer from class '%s'", transfererClass), e);
+            }
+        }
+
+
+        if (storageDir == null) {
+            storageDirectory = TestUtils.tempDirectory(ROOT_STORAGES_DIR_NAME + "-");
+
+            logger.debug("No storage directory specified, created temporary directory: {}",
+                    storageDirectory.getAbsolutePath());
+
+        } else {
+            storageDirectory = new File(storageDir + "/" + ROOT_STORAGES_DIR_NAME);
+            // NOTE: Provide the relative storage directory path to view the files in the same directory when running tests.
+            // storageDirectory = new File(new File("."), ROOT_STORAGES_DIR_NAME + "/" + storageDir);
```

Comment:
LocalTieredStorage implementation can be used in two modes: 

1. Integration test and
2. To bring Kafka in the local machine with `LocalTieredStorage` implementation as RSM for quick testing by adding `kafka-storage-<version>-test.jar` in the classpath, then enabling the `INCLUDE_TEST_JARS` flag before running the server

In mode 2, if the commented-out line#273 is enabled, then it creates the `storageDirectory` from the present kafka working directory which can cause confusion:

(eg)
Kafka is installed on `/opt/kafka` and `storageDirectory` is configured as `/tmp/kafka-tiered-storage`, then it will create it the directory in `/opt/kafka/kafka-tiered-storage/tmp/kafka-tiered-storage` instead of `/tmp/kafka-tiered-storage`. 

For the integration test, the `storageDirectory` won't be set so it uses the java temporary directory.

Reply:
Hmm.. should we keep it here? Or should we add more context in the comment? Obviously the comment is confusing now.

Reply:
Added the context as a comment on the code.